sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_mobile/HttpRequestMessageFactory.cs

```diff
@@ -39,9 +40,7 @@ namespace Amazon.Runtime
     public class HttpRequestMessageFactory : IHttpRequestFactory<HttpContent>
     {
         // This is the global cache of HttpClient for service clients that are using 
-        static readonly ReaderWriterLockSlim _httpClientCacheRWLock = new ReaderWriterLockSlim();
-        static readonly IDictionary<string, HttpClientCache> _httpClientCaches = new Dictionary<string, HttpClientCache>();
-
+        static readonly ConcurrentDictionary<string, HttpClientCache> _httpClientCaches = new ConcurrentDictionary<string, HttpClientCache>();
```

Comment:
Keys are case-sensitive - is that OK?

Reply:
That's likely more of a potential issue than an actual one. It's not a change in behaviour though.

Reply:
Yes you're right, I clearly forgot to properly look at the line that was deleted.